### DISCUSSION TITLE
How does swc use terser?

### DISCUSSION THREAD

**Discussion by mattfysh** [swc-project/swc#4820] [Q&A] [answered]
From the docs, I understand you use terser to implement minification:
https://swc.rs/docs/configuration/minification

However, when I look at the benchmarks on this site, swc is consistently one of the fastest options. Magnitudes faster than terser itself: https://github.com/privatenumber/minification-benchmarks

How is this possible?

**Comment by kwonoj** [comment, marked as answer]
SWC references terser's implementation at some places, SWC does NOT uses terser under the hood.

While there are some logics works similar, there are various architectual differences including runtime gives different performance numbers.

**Comment by mattfysh** [comment]
Thanks OJ! Does this mean SWC rewrote terser in Rust?

**Reply by kwonoj** [reply]
no.

**Reply by mattfysh** [reply]
Got it, I must have misinterpreted the minification docs when I saw the like-for-like API and assumed terser was used under the hood. Thanks for the clarification!

Also that's strange - I replied to you by replying to your email, and instead of adding it as a comment under your answer GH went and created a new answer 🤷
